xl-objects.go

```diff
@@ -416,29 +415,33 @@ func (xl xlObjects) PutObject(bucket string, object string, size int64, data io.
 }
 
 // isMultipartObject - verifies if an object is special multipart file.
-func isMultipartObject(storage StorageAPI, bucket, object string) (bool, error) {
+func isMultipartObject(storage StorageAPI, bucket, object string) bool {
 	_, err := storage.StatFile(bucket, pathJoin(object, multipartMetaFile))
 	if err != nil {
 		if err == errFileNotFound {
-			return false, nil
+			return false
 		}
-		return false, err
+		errorIf(err, "Failed to stat the file "+bucket+pathJoin(object, multipartMetaFile))
+		return false
 	}
-	return true, nil
+	return true
 }
 
 // Deletes and object.
 func (xl xlObjects) deleteObject(bucket, object string) error {
```

Comment:
DeleteObject should wrap deleteMultpartObject and deleteObject. Break up this function.